### PR TITLE
Fix Zmachine tests

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

- Tests were passing before because compute capacity class wasn't validated. Edited zmachine tests & validated its workload

image

### Changes

- Fix zmachine tests
- Add more tests
- Validate zmachine workload

### Related Issues

- #2458
- #3335

### Checklist

- [x] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
I think the tests are not enough, we can test every property on the `zmachine` workload.

we can even write json workload and try to cast it on the class to test the data type

**Comment by AhmedHanafy725** [comment]
please use better title for the PR

**Comment by Mahmoud-Emad** [comment]
Please resolve the conflicts

**Comment by zaelgohary** [comment]
> Please resolve the conflicts

Done.

**Comment by AhmedHanafy725** [comment]
> I think the tests are not enough, we can test every property on the `zmachine` workload.

still we can add more tests

**Review by Mahmoud-Emad — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]

File: packages/grid_client/tests/modules/zmachine.test.ts [hunk 1]
```diff
@@ -29,36 +35,136 @@ beforeAll(() => {
...
+  });
+
+  it("should fail validation for entering invalid flist", () => {
+    const invalidFlist = () => {
       zmachine.flist = "";
-      zmachine.entrypoint = "";
     };
+
+    expect(invalidFlist).toThrow();
+  });
```
Comment (AhmedHanafy725) [inline packages/grid_client/tests/modules/zmachine.test.ts]: you can test also if the flist is not a URL

File: packages/grid_client/tests/modules/zmachine.test.ts [hunk 2]
```diff
@@ -29,36 +35,136 @@ beforeAll(() => {
...
+    const invalidEntrypoint = () => (zmachine.entrypoint = undefined as any);
+
+    expect(invalidEntrypoint).toThrow();
+  });
+
+  it("should fail validation for entering invalid size", () => {
+    const invalidSize = () => (zmachine.size = 10 * 1024 ** 5);
+
+    expect(invalidSize).toThrow();
+  });
```
Comment (AhmedHanafy725) [inline packages/grid_client/tests/modules/zmachine.test.ts]: you can try negative numbers or fractions

**Review by AhmedHanafy725 — COMMENTED** [review]
you may think about more edge cases

File: packages/grid_client/tests/modules/zmachine.test.ts (hunk 2 above)
Comment (zaelgohary) [reply packages/grid_client/tests/modules/zmachine.test.ts]: I added more assertions and fixed the validation for flist & size in zmachine.

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]
